libnavigation-lifecycle/build.gradle

```diff
@@ -0,0 +1,39 @@
+apply plugin: 'com.android.library'
+apply plugin: 'kotlin-android'
+apply plugin: 'kotlin-kapt'
+
+android {
+    compileSdkVersion androidVersions.compileSdkVersion
+
+    compileOptions {
+        sourceCompatibility = JavaVersion.VERSION_1_8
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+
+    defaultConfig {
+        minSdkVersion androidVersions.minSdkVersion
+        missingDimensionStrategy("settings_visibility", "internal")
+
+        testInstrumentationRunner = "androidx.test.runner.AndroidJUnitRunner"
+        consumerProguardFiles 'proguard-rules.pro', "${rootDir}/proguard/proguard-project.pro"
+    }
+
+    buildFeatures {
+        viewBinding true
+    }
+}
+
+dependencies {
+    implementation project(":libnavigation-core")
```

Comment:
> Alternatively we can leave it as is for now while it's experimental, and move it to the core module when we think it's ready to become stable.

I'm in favor of this approach only because the jetpack lifecycle would need to be added as an `api` dependency in `core`. Maybe that's fine? But as a separate module we won't have to worry about it yet.

Reply:
I think that's fine, if we want to take`LifecycleOwner`s as arguments, we need to provide this dependency.

The only question is whether we expect this approach to be the recommended way of accessing the `MapboxNavigation` object instead of `MapboxNavigationProvider` and the answer seems to be yes, so I think it should live in `core` so that developers don't need to specify another dependency to get it.

Reply:
Ok sounds good to me, moved to core 👍